File: Helmut/2/solve.hs
--
import Data.Char (ord)

c [a,b] = (ord (head a)  - ord 'A', ord (head b) - ord 'X')

ev (a,b) = b + 1 + if (a,b) `elem` [(0,1),(1,2),(2,0)] then 6 
                             else if b == a then 3
                                            else 0
check = ev . c . words 

main = do
    print $ check "C Z"
    fmap (sum . map check . lines) $ readFile "input.txt" -- "small.txt"

File: Helmut/2/solve.rs
fn f(s: &str) -> i64 {
    match s {
        "A Y" => 6 + 2,
        "B Z" => 6 + 3,
        "C X" => 6 + 1,
        "A X" => 3 + 1,
        "B Y" => 3 + 2,
        "C Z" => 3 + 3,
        "A Z" => 0 + 3,
        "B X" => 0 + 1,
        "C Y" => 0 + 2,
        _ => unreachable!(),
    }
}

fn main() {
    let ss = include_str!("input.txt").lines().map(|s| f(s)).sum::<i64>();

    println!("{}", ss);
}
